### PR TITLE
Always call .filter before displaying section data

### PR DESCRIPTION
**Pull request by allait**
Since section fields can go through additional processing with Jinja we need to ensure we're always calling .filter before displaying them to the user so that we don't accidentally display template source.

### CODE REVIEW

File: app/main/views/frameworks.py [hunk 1]
```diff
@@ -911,14 +911,19 @@ def view_contract_variation(framework_slug, variation_slug):
                 {'question': form['accept_changes'].label.text, 'input_name': 'accept_changes'}
             ]
 
+    supplier_name = supplier_framework['declaration']['nameOfOrganisation']
+    variation_content = content_loader.get_message(framework_slug, variation_content_name).filter(
+        {'supplier_name': supplier_name}
+    )
```
Comment (pcraig3) [inline app/main/views/frameworks.py]: While quite a lot better than the old way of finding the supplier name, this still seems a bit ad-hoc.  
How are we supposed to think about it? Do we just pass in whatever we need as needed?

File: tests/app/main/test_frameworks.py [hunk 2]
```diff
@@ -3172,7 +3172,7 @@ def test_success_message_is_displayed_on_success(self, data_api_client):
         assert res.status_code == 200
         assert len(
             doc.xpath('//p[@class="banner-message"][contains(text(), "You have accepted the proposed changes.")]')
-        ) == 1
+        ) == 1, res.get_data(as_text=True)
```
Comment (allait) [inline tests/app/main/test_frameworks.py]: It's useful, otherwise that error message just says `0 != 1`

**Comment by pcraig3** [comment]
If the [pull request in utils](https://github.com/alphagov/digitalmarketplace-utils/pull/286) is approved, you could potentially pull that in as well.

**Comment by TheDoubleK** [comment]
If I rebase this now and add content 4.0.2 will it be good to go?

**Comment by pcraig3** [comment]
Hey Kev!  Hold on a sec before merging this.

**Pull request merged**